File: grouper/WPGGroupViewController.m
//
//  WPGGroupViewController.m
//  grouper
//
//  Created by 高磊 on 14-5-29.
//  Copyright (c) 2014年 WholePort. All rights reserved.
//

#import "WPGGroupViewController.h"
#import "WPGQianCell.h"

@interface WPGGroupViewController ()

@end

@implementation WPGGroupViewController
{
    NSMutableArray *opened;
    UIAlertView *alert;
}

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    opened = [NSMutableArray array];
    alert = [[UIAlertView alloc] initWithTitle:@"抽签已完成" message:@"是否重新抽签？" delegate:self cancelButtonTitle:@"取消" otherButtonTitles:@"重新抽签", nil];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (BOOL)prefersStatusBarHidden
{
    return YES;
}

- (void)done:(id)sender
{
    [self.navigationController popViewControllerAnimated:YES];
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

- (IBAction)returnStart:(id)sender
{
    [alert show];
}

#pragma mark - UIAlertViewDelegate

- (void)alertView:(UIAlertView *)alertView clickedButtonAtIndex:(NSInteger)buttonIndex
{
    if (buttonIndex != alertView.cancelButtonIndex) {
        [self dismissViewControllerAnimated:YES completion:nil];
    }
}

#pragma mark - UICollectionView

- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section
{
    return self.qian.count;
}

- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath
{
    WPGQianCell *cell = (WPGQianCell *)[collectionView dequeueReusableCellWithReuseIdentifier:@"Cell" forIndexPath:indexPath];
    BOOL hidden = YES;
    
    if ([opened containsObject:indexPath]) {
        hidden = NO;
    }
    
    
    [cell.label_qian setHidden:hidden];
    cell.label_qian.text = [self.qian objectAtIndex:indexPath.row];
    return cell;
}

- (void)collectionView:(UICollectionView *)collectionView didHighlightItemAtIndexPath:(NSIndexPath *)indexPath
{
    WPGQianCell *cell = (WPGQianCell *)[collectionView cellForItemAtIndexPath:indexPath];
    cell.contentView.backgroundColor = [UIColor brownColor];
}

- (void)collectionView:(UICollectionView *)collectionView didUnhighlightItemAtIndexPath:(NSIndexPath *)indexPath
{
    WPGQianCell *cell = (WPGQianCell *)[collectionView cellForItemAtIndexPath:indexPath];
    cell.contentView.backgroundColor = [UIColor colorWithRed:.5f green:.64f blue:.33f alpha:1.0f];
}

- (void)collectionView:(UICollectionView *)collectionView didSelectItemAtIndexPath:(NSIndexPath *)indexPath
{
    if (opened.count >= self.qian.count) {
        [self returnStart:self];
        return;
    }
    if ([opened containsObject:indexPath]) {
        return;
    }
    [opened addObject:indexPath];
    [collectionView reloadItemsAtIndexPaths:@[indexPath]];
    if (opened.count == self.qian.count) {
        self.navigationItem.rightBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemDone target:self action:@selector(done:)];
    }

}

@end

File: iGrouping/WPGGroupViewController.swift
//
//  WPGGroupViewController.swift
//  iGrouping
//
//  Created by 高磊 on 15/2/7.
//
//

import UIKit

let reuseIdentifier = "Cell"

class WPGGroupViewController: UICollectionViewController, UIAlertViewDelegate {
    
    var qian: NSArray!
    var opened: NSMutableArray!
    var alert: UIAlertView!

    override func viewDidLoad() {
        super.viewDidLoad()

        // Uncomment the following line to preserve selection between presentations
        // self.clearsSelectionOnViewWillAppear = false

        // Register cell classes
//        self.collectionView.registerClass(UICollectionViewCell.self, forCellWithReuseIdentifier: reuseIdentifier)

        // Do any additional setup after loading the view.
        opened = NSMutableArray()
        alert = UIAlertView(title: "抽签已完成", message: "是否重新抽签？", delegate: self, cancelButtonTitle: "取消", otherButtonTitles: "重新抽签")
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override func prefersStatusBarHidden() -> Bool {
        return true
    }
    
    func returnStart() {
        alert.show()
    }
    
    // MARK: - UIAlertViewDelegate
    
    func alertView(alertView: UIAlertView, clickedButtonAtIndex buttonIndex: Int) {
        if (buttonIndex != alertView.cancelButtonIndex) {
            self.dismissViewControllerAnimated(true, completion: { () -> Void in
                
            })
        }
    }

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using [segue destinationViewController].
        // Pass the selected object to the new view controller.
    }
    */

    // MARK: UICollectionViewDataSource

    override func collectionView(collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
        return qian.count
    }

    override func collectionView(collectionView: UICollectionView, cellForItemAtIndexPath indexPath: NSIndexPath) -> UICollectionViewCell {
        let cell = collectionView.dequeueReusableCellWithReuseIdentifier(reuseIdentifier, forIndexPath: indexPath) as WPGQianCell
    
        // Configure the cell
        var hidden: Bool = true
        if (opened.containsObject(indexPath)) {
            hidden = false
        }
        
        cell.label_qian.hidden = hidden
        cell.label_qian.text = qian.objectAtIndex(indexPath.row) as? String
    
        return cell
    }

    // MARK: UICollectionViewDelegate
    
    override func collectionView(collectionView: UICollectionView, didHighlightItemAtIndexPath indexPath: NSIndexPath) {
        collectionView.cellForItemAtIndexPath(indexPath)?.contentView.backgroundColor = UIColor.brownColor()
    }
    
    override func collectionView(collectionView: UICollectionView, didUnhighlightItemAtIndexPath indexPath: NSIndexPath) {
        collectionView.cellForItemAtIndexPath(indexPath)?.contentView.backgroundColor = UIColor(red: 0.5, green: 0.65, blue: 0.33, alpha: 1.0)
    }
    
    override func collectionView(collectionView: UICollectionView, didSelectItemAtIndexPath indexPath: NSIndexPath) {
        if (opened.count >= qian.count) {
            self.returnStart()
            return
        }
        
        if (opened.containsObject(indexPath)) {
            return
        }
        
        opened.addObject(indexPath)
        
        collectionView.reloadItemsAtIndexPaths([indexPath])
    }

    /*
    // Uncomment this method to specify if the specified item should be highlighted during tracking
    override func collectionView(collectionView: UICollectionView, shouldHighlightItemAtIndexPath indexPath: NSIndexPath) -> Bool {
        return true
    }
    */

    /*
    // Uncomment this method to specify if the specified item should be selected
    override func collectionView(collectionView: UICollectionView, shouldSelectItemAtIndexPath indexPath: NSIndexPath) -> Bool {
        return true
    }
    */

    /*
    // Uncomment these methods to specify if an action menu should be displayed for the specified item, and react to actions performed on the item
    override func collectionView(collectionView: UICollectionView, shouldShowMenuForItemAtIndexPath indexPath: NSIndexPath) -> Bool {
        return false
    }

    override func collectionView(collectionView: UICollectionView, canPerformAction action: Selector, forItemAtIndexPath indexPath: NSIndexPath, withSender sender: AnyObject?) -> Bool {
        return false
    }

    override func collectionView(collectionView: UICollectionView, performAction action: Selector, forItemAtIndexPath indexPath: NSIndexPath, withSender sender: AnyObject?) {
    
    }
    */

}
